File: back/Finansy.Domain/Entities/Administrador.cs
using Finansy.Domain.Contracts;
using Finansy.Domain.Validators;
using FluentValidation.Results;

namespace Finansy.Domain.Entities;

public class Administrador : Entity, ISoftDelete, IAggregateRoot
{
    public string Nome { get; set; } = null!;
    public string Senha { get; set; } = null!;
    public string? Telefone { get; set; }
    public string Cpf { get; set; } = null!;
    public string Email { get; set; } = null!;
    public bool Desativado { get; set; }

    public List<Unidade> Unidades { get; set; } = new();

    public override bool Validate(out ValidationResult validationResult)
    {
        validationResult = new AdminValidator().Validate(this);
        return validationResult.IsValid;
    }
}

File: angular/finansy/src/app/pages/admin/models/administrador.ts
export interface AdministradorViewModel {
  id: number;
  nome: string;
  email: string;
  telefone?: string;
  cpf: string;
  desativado: boolean;
}

export interface CadastrarViewModel {
  nome: string;
  email: string;
  senha: string;
  telefone?: string;
  cpf: string;
}

export interface LoginViewModel {
  email: string;
  senha: string;
}

export interface AtualizarViewModel {
  id: number;
  nome: string;
  email: string;
  telefone?: string;
  cpf: string;
}
